src/wallet/hardwaresigner/signer.rs

```diff
@@ -0,0 +1,55 @@
+//! HWI Signer
+//!
+//! This module contains a simple implementation of a Custom Signer for rust-hwi
+
+use bitcoin::{
+    psbt::PartiallySignedTransaction,
+    secp256k1::{All, Secp256k1},
+    util::bip32::Fingerprint,
+};
+use hwi::{
+    error::Error,
+    types::{HWIChain, HWIDevice},
+    HWIClient,
+};
+
+use crate::signer::{SignerCommon, SignerError, SignerId, TransactionSigner};
+
+#[derive(Debug)]
+/// Custom Signer for Hardware Wallets
+/// This ignores `sign_options` and leaves the decisions upto the hardware wallet.
+pub struct HWISigner {
+    fingerprint: Fingerprint,
+    client: HWIClient,
+}
+
+impl HWISigner {
+    /// Create a instance from the specified device and chain
+    pub fn from_device(device: &HWIDevice, chain: HWIChain) -> Result<HWISigner, Error> {
+        let client = HWIClient::get_client(device, false, chain)?;
+        Ok(HWISigner {
+            fingerprint: device.fingerprint,
+            client,
+        })
+    }
+}
+
+impl SignerCommon for HWISigner {
+    fn id(&self, _secp: &Secp256k1<All>) -> SignerId {
+        SignerId::Fingerprint(self.fingerprint)
+    }
+}
+
+/// This implementation ignores `sign_options`
+impl TransactionSigner for HWISigner {
+    fn sign_transaction(
+        &self,
+        psbt: &mut PartiallySignedTransaction,
+        _sign_options: &crate::SignOptions,
+        _secp: &crate::wallet::utils::SecpCtx,
+    ) -> Result<(), SignerError> {
+        psbt.combine(self.client.sign_tx(psbt)?.psbt)
+            .expect("Failed to combine HW signed psbt with passed PSBT");
```

Comment:
Is it safe to have an "expect" here? In which cases can the psbt combine fail?

Reply:
Looking at the code for `psbt.combine()`, I could find 2 possible errors that can be returned, (1) when the 2 psbts describe different txs (shouldn't happen as the hardware wallet should only sign the tx) and (2) `CombineInconsistentKeySources ` which, AFAIK, shouldn't happen unless the user wipes and resets the device _while_ he is signing a transaction XD.

Reply:
Thanks for looking into this!